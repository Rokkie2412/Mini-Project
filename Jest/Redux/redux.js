export const ADD = 'ADD'
export const DELETE = 'DELETE'

export const addArray = (data) =>({
    type:ADD,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch(action.type){
        case ADD:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE:
            return{
                ...state,arr:state.arr.filter(arr => arr.id !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer