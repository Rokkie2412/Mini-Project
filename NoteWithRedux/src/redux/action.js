import { ADDARRAY } from "./actionType";

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})