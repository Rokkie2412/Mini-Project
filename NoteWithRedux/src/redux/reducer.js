import { ADDARRAY } from "./actionType";

const initialState = {
    arr:[],
    
};

export const mainReducer = (state = initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
           
        default:
            return state;
    }
}