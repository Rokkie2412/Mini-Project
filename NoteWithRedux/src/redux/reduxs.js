import remove from "lodash.remove"
export const ADDARRAY = 'ADDARRAY'
export const DELETENOTE = 'DELETENOTE'
export const EDITNOTE = 'EDITNOTE'

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})

export const deleteFromArray = (title) => ({
    type:DELETENOTE,
    payload:title
})
export const EditNote = (title) => ({
    type:EDITNOTE,
    payload:title,
})



const initialState = []

export const mainReducer = (state=initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return[
                ...state,action.payload
            ]
        case DELETENOTE:
            const deleteNewArray = remove(state, obj => {
                return obj.title != action.payload
            })
            return deleteNewArray
        case EDITNOTE:
            // const filterDelete = map(state,obj =>{
            //     return obj.title == action.payload
            // })
            const filtered = state.filter((state,obj)=>{
                if(obj.title === action.payload){
                    var tamp = obj.title
                    console.log(tamp);
                }
            })
        default:
            return state;
    }
}

export default mainReducer