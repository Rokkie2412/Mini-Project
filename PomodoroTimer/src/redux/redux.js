
/**
 * @format
 * @flow
 */
export const EXCHAHNGETOSECONDS : string = 'EXCHANGETOSECOND'
export const EXCHAHNGETOSECONDBREAK :string = 'EXCHANGETOSECONDBREAK'
export const EXCHAHNGETOSECONDLONGBREAK :string = 'EXCHANGETOSECONDLONGBREAK'
export const GETDATA:string = 'GETDATA'

export const exchangetosecwork = (hour:string,minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDS,
    payloadhourwork : hour,
    payloadminutework : minute,
    payloadsecondwork : second
})

export const exchangetosecBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDBREAK,
    payloadminutebreak : minute,
    payloadsecondbreak : second
})

export const exchangetosecLongBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDLONGBREAK,
    payloadminutelongbreak : minute,
    payloadsecondlongbreak : second
})

let initialState : Object  ={
    resultWorkinSecond : 0,
    resultBreakinSecond : 0,
    resultLongBreakinSecond : 0
}

export const mainReducer = (state:Object=initialState, action:function):Function =>{
    switch (action.type) {
        case EXCHAHNGETOSECONDS:
            
            const hourTosecond : number = action.payloadhourwork * 3600
            const minuteTosecond : number = action.payloadminutework * 60
            const res : number = hourTosecond+minuteTosecond+action.payloadsecondwork
            return {
                ...state,
                resultWorkinSecond:res
            }
        case EXCHAHNGETOSECONDBREAK:
            const minuteTosecondBreak : number= action.payloadminutebreak * 60
            const resBreak : number = minuteTosecondBreak+action.payloadsecondbreak
            return {
                ...state,
                resultBreakinSecond:resBreak
            }
        case EXCHAHNGETOSECONDLONGBREAK:
            const minuteToSecondLongBreak : number= action.payloadminutelongbreak * 60
            const resLongBreak : number = minuteToSecondLongBreak+action.payloadsecondlongbreak
            return{
                ...state,
                resultLongBreakinSecond:resLongBreak
            }
        default:
            return state
    }
}

export default mainReducer