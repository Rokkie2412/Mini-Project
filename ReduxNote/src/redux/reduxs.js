import remove from "lodash.remove";
export const ADD_NOTE = 'ADDNOTE'
export const DELETE_NOTE = 'DELETE_NOTE'

export const addArray = (data) =>({
    type:ADD_NOTE,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE_NOTE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch (action.type) {
        case ADD_NOTE:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE_NOTE:
            // const deleteNewArray = remove(state=initialState,obj =>{
            //     return obj.title != action.payload
            // })
            // return state.arr = deleteNewArray
            return{
                ...state,arr:state.arr.filter(arr => arr.title !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer