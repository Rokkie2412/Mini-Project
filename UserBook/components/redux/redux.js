export const ADD_CONTACT = 'ADD_CONTACT'
export const SHOW_CONTACT = 'SHOW_CONTACT'
export const EDITCONTACT = 'EDITCONTACT'

const ApiUrl = 'https://simple-contact-crud.herokuapp.com/contact'


export const addContact = (firstName,lastName,age,image) =>({
    type:ADD_CONTACT,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

export const editContact = (itemId,firstName,lastName,age,image) => ({
    type:EDITCONTACT,
    payloadID : itemId,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

let initialState = {
    arr:[],
}

export const mainReducer = (state=initialState, action) => {
    switch (action.type) {
        case ADD_CONTACT:
            const postContact = () =>{
                fetch('https://simple-contact-crud.herokuapp.com/contact',{
                    method:'POST',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(postContact())
        case EDITCONTACT :
            const editContact = () =>{
                const tempid = action.payloadID 
                
                fetch(`https://simple-contact-crud.herokuapp.com/contact/${tempid}`,{
                    method:'PUT',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                 }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(editContact())
        default:
            return state
    }
}