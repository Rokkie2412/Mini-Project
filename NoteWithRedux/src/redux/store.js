import { createStore } from "redux";
import {mainReducer} from "./reduxs";

export const store = createStore(mainReducer)