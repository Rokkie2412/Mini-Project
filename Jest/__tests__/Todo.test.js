import React from "react";
import { create,act } from "react-test-renderer";
import Mentah from '../Mentahan'
import Todo from "../TodoList";
    import { createStore } from "redux";
import { Provider } from "react-redux";
import mainReducer from "../Redux/redux";
import Icon from 'react-native-vector-icons/MaterialIcons'

jest.mock(
  'react-native-vector-icons/MaterialIcons',
  () => 'MockedMaterialIcons',
);

const store=createStore(mainReducer,{status:'default status'})
const tree = create(
    <Provider store={store}>
        <Todo/>  
    </Provider>
)

// const tree = create(<RandomScreen/>)

test('Testing a component' ,()=>{
    expect(tree).toMatchSnapshot()
})

const treeMentah = create(<Mentah/>)
test('Mentah Screen',()=>{
    expect(treeMentah).toMatchSnapshot()
})


