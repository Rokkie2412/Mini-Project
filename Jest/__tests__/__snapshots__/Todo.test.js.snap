// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mentah Screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
      }
    }
    testID="myButton"
  >
    <Text
      style={
        Object {
          "backgroundColor": "#333",
          "borderRadius": 50,
          "color": "#fff",
          "padding": 20,
        }
      }
    >
      Press Me
    </Text>
  </View>
</View>
`;

exports[`Testing a component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "color": "#333",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginVertical": 12,
        "textAlign": "center",
      }
    }
  >
    My Todo List
  </Text>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Start Todo List Here"
      placeholderTextColor="#fff"
      style={
        Object {
          "backgroundColor": "#333",
          "borderRadius": 40,
          "color": "#fff",
          "fontSize": 16,
          "marginHorizontal": 12,
          "marginVertical": 10,
          "padding": 17,
          "paddingHorizontal": 25,
          "width": "90%",
        }
      }
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="myButton"
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "backgroundColor": "#333",
            "borderRadius": 50,
            "color": "#fff",
            "fontSize": 35,
            "marginVertical": 10,
            "padding": 15,
            "textAlign": "center",
            "width": 70,
          }
        }
        testID="myText"
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
